### PR TITLE
15.0 mig sale restricted qty

### PR DESCRIPTION
**Pull request by thinkwelltwd**
Migrate module as well as improve grammar and spelling
Optimize test by setting up test class once
Remove product_product_views.xml view since the product_template_views.xml view adds needed fields

### CODE REVIEW

File: sale_restricted_qty/__manifest__.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 {
     "name": "Sale order min quantity",
-    "version": "14.0.1.1.1",
+    "version": "15.0.1.1.1",
```
Comment (rousseldenis) [inline sale_restricted_qty/__manifest__.py]: Please reset to 15.0.1.0.0

File: sale_restricted_qty/tests/test_sale.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import odoo.tests.common as common
+from odoo.exceptions import ValidationError
+from odoo.tests import tagged
+
+
+@tagged("post_install", "-at_install")
+class TestSaleOrderLineMinQty(common.TransactionCase):
+    def setUp(self):
```
Comment (rousseldenis) [inline sale_restricted_qty/tests/test_sale.py]: I would have changed this to classmethod and add this line to improve performances:

`cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))`

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review

File: sale_restricted_qty/__manifest__.py (hunk 1 above)
Comment (thinkwelltwd) [reply sale_restricted_qty/__manifest__.py]: done...

File: sale_restricted_qty/tests/test_sale.py (hunk 2 above)
Comment (thinkwelltwd) [reply sale_restricted_qty/tests/test_sale.py]: Done...

**Comment by thinkwelltwd** [comment]
Thanks for the review! Should be ready to merge now.

File: sale_restricted_qty/models/product_restricted_qty_mixin.py [hunk 3]
```diff
@@ -11,38 +11,45 @@ class ProductMinMultipleMixin(models.AbstractModel):
...
-        "Product/product Template/product category/parent categoroies ",
+        "Hierarchy is in this order: "
+        "Product/product Template/product category/parent categories ",
         digits="Product Unit of Measure",
     )
     manual_sale_multiple_qty = fields.Float(
-        string="Multiple Sale Qty", digits="Product Unit of Measure"
+        string="Multiple Sale Qty",
+        digits="Product Unit of Measure",
+        recursive=True,
```
Comment (rousseldenis) [inline sale_restricted_qty/models/product_restricted_qty_mixin.py]: You should remove recursive property on non-computed fields

**Comment by rafaelbn** [comment]
We need here functional reviews for sale_restricted_qty module

/ocabot rebase

**Comment by rafaelbn** [comment]
Hello @rousseldenis , your comments have been attended!

@thinkwelltwd please review why pre-commit is failing! Thank you! 👍🏼

**Review by rousseldenis — COMMENTED** [review]
@thinkwelltwd Please fix pre-commit

File: sale_restricted_qty/models/product_restricted_qty_mixin.py (hunk 3 above)
Comment (thinkwelltwd) [reply sale_restricted_qty/models/product_restricted_qty_mixin.py]: `recursive` property now removed. Sorry for missing your comments till now.

**Comment by thinkwelltwd** [comment]
Could @rousseldenis or someone else please review this PR? The tests are all passing and it represents quite a lot of cleanup to the module. :pray:

**Review by rousseldenis — APPROVED** [review]

**Comment by thinkwelltwd** [comment]
All requested changes have been made and approved. The code has been used in production for weeks.

Could this PR be merged? :pray:

**Comment by rafaelbn** [comment]
Please, could you review in runbot showing how this works? 😄 

I've tested and it doesn't look like working

imagen

**Comment by alexey-pelykh** [comment]
Maybe we should continue in #2992?

@rafaelbn "force" in this module means "allow force override", so you need to have that unchecked. I'm addressing all of that UX in #2997

**Comment by rousseldenis** [comment]
@thinkwelltwd Do you mind if we switch to #2992 ?

**Comment by thinkwelltwd** [comment]
> @thinkwelltwd Do you mind if we switch to #2992 ?

Sure, go ahead. Thanks for asking. :smile:

**Comment by thinkwelltwd** [comment]
Superseded by #2992

**Pull request closed without merging**